Update ml-commons metrics to add agent details (#11163)
opensearch-project/documentation-website: _monitoring-your-cluster/metrics/getting-started.md

Introduced 3.1
{: .label .label-purple }

OpenSearch provides enhanced observability for [machine learning (ML)]({{site.url}}{{site.baseurl}}/ml-commons-plugin/) workflows. Metrics related to ML operations are pushed directly to the core metrics registry, giving you improved visibility into model usage and performance. Additionally, every 5 minutes, a periodic job collects and exports state data, helping you monitor the health and activity of your ML workloads over time.

> The static collector job captures details about different types of models and agents created:

Reviewer: ```suggestion
The static collector job captures the following metrics about different types of models and agents created:
```

> 1. **Models** - Including deployment type (remote, pretrained, custom), service provider, algorithm, model name, and model type.

Reviewer: ```suggestion
- **Models**: Deployment type (remote, pretrained, or custom), service provider, algorithm, model name, and model type.
```

> 2. **Agents** - Including LLM interface, model deployment, service provider, model type, memory type, and model identifier.

Reviewer: ```suggestion
- **Agents**: LLM interface, model deployment type, service provider, model type, memory type, and model identifier.
```

> Additional metrics will be added in future releases.

Reviewer: ```suggestion
```

> Example model metrics captured:

Reviewer: ```suggestion
The following is an example of captured model metrics:

```

```
{is_hidden=false, service_provider=openai, model=gpt-3.5-turbo, type=llm, deployment=remote, algorithm=REMOTE}
```

> Example agent metrics captured:

Reviewer: ```suggestion
The following is an example of captured agent metrics:

```

Introduced 3.1
{: .label .label-purple }

OpenSearch provides enhanced observability for [machine learning (ML)]({{site.url}}{{site.baseurl}}/ml-commons-plugin/) workflows. Metrics related to ML operations are pushed directly to the core metrics registry, giving you improved visibility into model usage and performance. Additionally, every 5 minutes, a periodic job collects and exports state data, helping you monitor the health and activity of your ML workloads over time.

The static collector job captures the following metrics about different types of models and agents created:

- **Models**: Deployment type (remote, pretrained, or custom), service provider, algorithm, model name, and model type.

> - **Agents**: LLM interface, model deployment type, service provider, model type, memory type, and model identifier.

Reviewer 2: ```suggestion
- **Agents**: LLM interface, model deployment type, service provider, model type, memory type, and model identifier
```
